examples/microwave-oven-app/microwave-oven-common/include/microwave-oven-device.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include "microwave-oven-control-delegate.h"
+#include "operational-state-delegate.h"
+#include <app/clusters/microwave-oven-control-server/microwave-oven-control-server.h>
+#include <app/clusters/operational-state-server/operational-state-server.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class MicrowaveOvenDevice
+{
+private:
+    MicrowaveOvenControl::MicrowaveOvenControlDelegate mMicrowaveOvenControlDelegate;
+    MicrowaveOvenControl::Instance mMicrowaveOvenControlInstance;
+
+    OperationalState::OperationalStateDelegate mOperationalStateDelegate;
+    OperationalState::Instance mOperationalStateInstance;
+
+public:
+    /**
+     * This class is responsible for initialising all the microwave oven device clusters and manging the interactions between them
+     * as required by the specific "business logic". See the state machine diagram.
+     * @param aClustersEndpoint The endpoint ID where all the microwave oven clusters exist.
+     */
+    explicit MicrowaveOvenDevice(EndpointId aClustersEndpoint) :
+        mMicrowaveOvenControlDelegate(),
+        mMicrowaveOvenControlInstance(&mMicrowaveOvenControlDelegate, aClustersEndpoint, MicrowaveOvenControl::Id),
+        mOperationalStateDelegate(), mOperationalStateInstance(&mOperationalStateDelegate, aClustersEndpoint, OperationalState::Id)
+    {
+        // set callback functions
+        mMicrowaveOvenControlDelegate.SetMicrowaveOvenControlSetCookingParametersCallback(
+            &MicrowaveOvenDevice::HandleMicrowaveOvenSetCookingParametersCallback, this);
+        mMicrowaveOvenControlDelegate.SetMicrowaveOvenControlAddMoreTimeCallback(
+            &MicrowaveOvenDevice::HandleMicrowaveOvenAddMoreTimeCallback, this);
+        mOperationalStateDelegate.SetOpStatePauseCallback(&MicrowaveOvenDevice::HandleMicrowaveOvenOpStatePasueCallback, this);
```

Comment:
typo in `HandleMicrowaveOvenOpStatePasueCallback` --> `HandleMicrowaveOvenOpStatePauseCallback`